core/src/test/kotlin/org/jetbrains/kotlinx/dataframe/api/replace.kt

```diff
@@ -13,4 +15,61 @@ class ReplaceTests {
         conv.columnNames() shouldBe listOf("b")
         conv.columnTypes() shouldBe listOf(typeOf<Double>())
     }
+
+    @Test
+    fun `unfold primitive`() {
+        val a by columnOf("123")
+        val df = dataFrameOf(a)
+
+        val conv = df.replace { a }.unfold {
+            "b" from { it }
+            "c" from { DataRow.readJsonStr("""{"prop": 1}""") }
+        }
+
+        val b = conv["a"]["b"]
+        b.type() shouldBe typeOf<String>()
+        b.values() shouldBe listOf("123")
+
+        val c = conv["a"]["c"]["prop"]
+        c.type() shouldBe typeOf<Int>()
+        c.values() shouldBe listOf(1)
+    }
+
+    @Test
+    fun `unfold properties`() {
+        val col by columnOf(A("1", 123, B(3.0)))
+        val df1 = dataFrameOf(col)
+        val conv = df1.replace { col }.unfold(maxDepth = 2)
```

Comment:
Not specifying maxDepth now breaks, while it worked before.
Try running `df1.replace { col }.with { it.unfold() }` before the PR and after it.
It works before, but now it gives: `java.lang.UnsupportedOperationException: Can not get nested column 'd' from ValueColumn 'bb'`